rust/aura-core/src/cache.rs

```diff
@@ -264,7 +264,20 @@ pub fn missing_tarballs<'a>(
 pub fn is_package(path: &Path) -> bool {
     path.to_str()
         .map(|p| {
-            p.ends_with(".pkg.tar.zst") || p.ends_with(".pkg.tar.xz") || p.ends_with(".pkg.tar")
+            [
+                ".pkg.tar.gz",
+                ".pkg.tar.bz2",
+                ".pkg.tar.xz",
+                ".pkg.tar.zst",
+                ".pkg.tar.lrz",
+                ".pkg.tar.lzo",
+                ".pkg.tar.Z",
+                ".pkg.tar.lz4",
+                ".pkg.tar.lz",
+                ".pkg.tar",
+            ]
+            .iter()
+            .any(|ext| p.ends_with(ext))
```

Comment:
This function is used in a number of places and in some cases may be called thousands of times. I'd like to avoid the allocation of an `Iter` and the `O(n)` lookup if possible, but for that we'd need a `HashSet` or something, yet can't allocate one statically.

Another approach would be to just have a large chain of `||` calls, with the most likely candidates first. Any thoughts?

Reply:
I checked on Godbolt and all the iterator stuff seems to be inlined. [Here's the code generated with `any()`](https://godbolt.org/z/nE99Efjb6) and [here's the code with a `||` chain](https://godbolt.org/z/1zYxP7WT3). The order is a little different but the basic operations are the same.

I can move `.pkg.tar`, `.pkg.tar.zst`, and `.pkg.tar.xz` to the top, though. That does affect the order in which comparisons are made in the assembly so it'll be a bit faster to return early on those cases.